test/test_nn.py

```diff
@@ -13985,6 +14035,33 @@ def test_upsamplingNearest2d(self, device):
 
                     self.assertEqual(a_cuda.grad, a_cpu.grad)
 
+        # checks data copy if output_size == input_size
```

Comment:
Why does this have to be duplicated? Is there a way to maybe create a helper function to avoid the duplication here?

Reply:
Yes, you are right, we can refactor these subtests. As I'm still inspecting the root cause of the issue trying to avoid specific handling for these cases: scale = 1 and 2, I think once done I can propose more appropriate test cases.